lib/omnibus/software.rb

```diff
@@ -693,16 +703,34 @@ def version_for_cache
     # @return [Fetcher]
     #
     def fetcher
-      @fetcher ||= if source
+      @fetcher ||= case source_type
+                   when :url
+                     NetFetcher.new(self)
+                   when :git
+                     GitFetcher.new(self)
+                   when :path
+                     PathFetcher.new(self)
+                   when :project_local
+                     NullFetcher.new(self)
+                   end
+    end
+
+    #
+    # The type of source specified for this software defintion.
+    #
+    # @return [String]
+    #
+    def source_type
```

Comment:
This feels like it is an attribute of the fetcher: `some_software_instance.fetcher.type`.